oxidation/libparsec/crates/client_types/src/local_device_file.rs

```diff
@@ -142,176 +134,360 @@ impl AvailableDevice {
             .map(|x| x.as_ref())
             .unwrap_or_else(|| self.device_id.device_name().as_ref())
     }
+}
 
-    /// For the legacy device files, the slug is contained in the device filename
-    fn load(key_file_path: PathBuf) -> LocalDeviceResult<Self> {
-        let (ty, organization_id, device_id, human_handle, device_label, slug) =
-            match DeviceFile::load(&key_file_path)? {
-                DeviceFile::Password(device) => (
-                    DeviceFileType::Password,
-                    device.organization_id,
-                    device.device_id,
-                    device.human_handle,
-                    device.device_label,
-                    // Handle legacy device
-                    match device.slug {
-                        Some(slug) => slug,
-                        None => {
-                            let slug = key_file_path
-                                .file_stem()
-                                .expect("Unreachable because deserialization succeed")
-                                .to_str()
-                                .expect("It may be unreachable")
-                                .to_string();
-
-                            if LocalDevice::load_slug(&slug).is_err() {
-                                return Err(LocalDeviceError::InvalidSlug);
-                            }
+#[cfg(not(target_arch = "wasm32"))]
+pub use native::*;
+#[cfg(target_arch = "wasm32")]
+pub use wasm32::*;
+
+#[cfg(not(target_arch = "wasm32"))]
```

Comment:
For readability:

- move the native & wasm32 modules in there own files
- move the test part in there own file, ideally in the `tests` folder (so in integration test, remember `integration test > unit test` given 1) tests are not too dependent on implementation details and 2) this limit the noise in the src folder).

Reply:
I don't want to publish the `_DEVICE` in public interface

Reply:
Fine, let's say this test very correlated with `_DEVICE`, not too big, and help understand what this blob contains. So it's ok to have it within the same file 👍